Remotion/Security/UnitTests/SecurityClientTests/SecurityClientTestHelper.cs

```diff
@@ -17,44 +17,42 @@
 using System;
 using System.Collections.Generic;
 using System.Reflection;
+using Moq;
 using Remotion.Reflection;
 using Remotion.Security.Metadata;
 using Remotion.Security.UnitTests.SampleDomain;
-using Rhino.Mocks;
 
 namespace Remotion.Security.UnitTests.SecurityClientTests
 {
   public class SecurityClientTestHelper
   {
-    private readonly MockRepository _mocks;
-    private readonly ISecurityPrincipal _userStub;
-    private readonly ISecurityProvider _mockSecurityProvider;
-    private readonly IPermissionProvider _mockPermissionReflector;
-    private readonly IObjectSecurityStrategy _mockObjectSecurityStrategy;
-    private readonly IFunctionalSecurityStrategy _mockFunctionalSecurityStrategy;
-    private readonly IMemberResolver _mockMemberResolver;
-    private readonly IPrincipalProvider _stubPrincipalProvider;
+    private readonly Mock<ISecurityPrincipal> _userStub;
+    private readonly Mock<ISecurityProvider> _mockSecurityProvider;
+    private readonly Mock<IPermissionProvider> _mockPermissionReflector;
+    private readonly Mock<IObjectSecurityStrategy> _mockObjectSecurityStrategy;
+    private readonly Mock<IFunctionalSecurityStrategy> _mockFunctionalSecurityStrategy;
+    private readonly Mock<IMemberResolver> _mockMemberResolver;
+    private readonly Mock<IPrincipalProvider> _stubPrincipalProvider;
     private readonly SecurableObject _securableObject;
 
     public SecurityClientTestHelper ()
     {
-      _mocks = new MockRepository();
-      _mockSecurityProvider = _mocks.StrictMock<ISecurityProvider>();
-      _mockPermissionReflector = _mocks.StrictMock<IPermissionProvider>();
-      _mockObjectSecurityStrategy = _mocks.StrictMock<IObjectSecurityStrategy>();
-      _mockFunctionalSecurityStrategy = _mocks.StrictMock<IFunctionalSecurityStrategy>();
-      _mockMemberResolver = _mocks.StrictMock<IMemberResolver>();
-      _userStub = _mocks.Stub<ISecurityPrincipal>();
-      SetupResult.For (_userStub.User).Return ("user");
-      _stubPrincipalProvider = _mocks.Stub<IPrincipalProvider>();
-      SetupResult.For (_stubPrincipalProvider.GetPrincipal()).Return (_userStub);
-
-      _securableObject = new SecurableObject (_mockObjectSecurityStrategy);
+      _mockSecurityProvider = new Mock<ISecurityProvider> (MockBehavior.Strict);
+      _mockPermissionReflector = new Mock<IPermissionProvider> (MockBehavior.Strict);
+      _mockObjectSecurityStrategy = new Mock<IObjectSecurityStrategy> (MockBehavior.Strict);
+      _mockFunctionalSecurityStrategy = new Mock<IFunctionalSecurityStrategy> (MockBehavior.Strict);
+      _mockMemberResolver = new Mock<IMemberResolver> (MockBehavior.Strict);
+      _userStub = new Mock<ISecurityPrincipal>();
+      _userStub.Setup (_ => _.User).Returns ("user").Verifiable();
```

Comment:
Why did the tooling make this verifyable?

Reply:
The original line was `SetupResult.For (_userStub.User).Return ("user");`
`SetupResult.For` is converted to `Expect (...)`, which is later converted to `Setup (...).Verifiable (...)`.
So it does not matter if `SetupResult.For` contains a `Stub` or a `Mock` it will always be verifiable. However in contrast to a `Mock` a `Stub` is not automatically verified in a converted `VerifyAll` statement, so even though the converted `Stub` is verifiable it will not be verfified.

Reply:
okay...I think we once discussed this as Expect is always acting as a Stub if there's a stub in there. So, the conversion could take that into account. Anyhow... we can drop the Verifyable here and below.